lab/lab2/js/part2-app-state.js

```diff
@@ -33,19 +33,30 @@
 ===================== */
 
 // Use the data source URL from lab 1 in this 'ajax' function:
-var downloadData = $.ajax("http://");
+var downloadData = $.ajax("https://raw.githubusercontent.com/CPLN692-MUSA611-Open-Source-GIS/datasets/master/json/philadelphia-crime-snippet.json");
 
 // Write a function to prepare your data (clean it up, organize it
 // as you like, create fields, etc)
-var parseData = function() {};
+var parseData = function(rawdata) {
+  return JSON.parse(rawdata)
+};
 
 // Write a function to use your parsed data to create a bunch of
 // marker objects (don't plot them!)
-var makeMarkers = function() {};
+var makeMarkers = function(parsed) {
+  result=[]
+  parsed.forEach(add_new)
+  function add_new(item) {
+    result.push([item.Lat,item.Lng])
+  }
+  return result
+};
 
 // Now we need a function that takes this collection of markers
 // and puts them on the map
-var plotMarkers = function() {};
+var plotMarkers = function(markers) {
+  _.each(markers,L.marker(marker).addTo(map))
```

Comment:
Again no markers showed up, so I couldn't finish the rest of the lab

Reply:
This is, again, going to be a situation in which a function  is what's required. `marker` is undefined here. I think what you want is: `_.each(markers, function(marker) { L.marker(marker).addTo(map) })`